internal/sync/sync.go

```diff
@@ -179,32 +179,45 @@ func (s *Synchronizer) sync() error {
 func (s *Synchronizer) Status() *types.SyncStatus {
 	latestBlockSaved := float64(s.syncManager.GetLatestBlockSaved())
 
+	startingBlockHash := new(felt.Felt)
+	startingBlockNumber := uint64(math.Min(float64(s.startingBlockNumber), latestBlockSaved))
+
+	currentBlockHash := new(felt.Felt)
 	latestBlockNumber := uint64(math.Min(float64(s.latestBlockNumberSynced), latestBlockSaved))
 
-	block, err := s.blockManager.GetBlockByNumber(latestBlockNumber)
+	highestBlockHash := new(felt.Felt)
+	highestBlockNumber := uint64(0)
+
+	startingBlock, err := s.blockManager.GetBlockByNumber(startingBlockNumber)
 	if err != nil {
-		return nil
+		if startingBlockNumber != 0 {
+			return nil
+		}
+		startingBlockHash = new(felt.Felt).SetHex("0x0")
```

Comment:
I think will be more clear if the error is checked using `errors.Is(...)` with the [ErrNotFound](https://github.com/NethermindEth/juno/blob/stdevMac/fix_starknet_syncing/internal/db/mdbx.go#L15)

Reply:
I am now checking with there Error, and also checking that is on the block 0, that is the only edge case, any other condition, is wrong